File: app/src/main/java/com/ayvytr/androidcustomviews/testactivity/ClearableEditTextActivity.java
package com.ayvytr.androidcustomviews.testactivity;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.TextView;

import com.ayvytr.androidcustomviews.R;
import com.ayvytr.customview.custom.text.ClearableEditText;

public class ClearableEditTextActivity extends AppCompatActivity {

    private ClearableEditText et;
    private TextView tvFocus;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_clearable_edit_text);
        initView();
    }

    private void initView() {
        et = findViewById(R.id.et);
        tvFocus = findViewById(R.id.tvFocus);
        et.setOnFocusChangeListener(new View.OnFocusChangeListener() {

            @Override
            public void onFocusChange(View v, boolean hasFocus) {
                tvFocus.setText("是否获得焦点:" + hasFocus);
            }
        });
    }
}

File: app/src/main/java/com/ayvytr/androidcustomviews/testactivity/ClearableEditTextActivity.kt
package com.ayvytr.androidcustomviews.testactivity

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.ayvytr.androidcustomviews.R
import com.ayvytr.customview.custom.text.ClearableEditText
import kotlinx.android.synthetic.main.activity_clearable_edit_text.*

class ClearableEditTextActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_clearable_edit_text)
        initView()
    }

    private fun initView() {
        setTitle(ClearableEditText::class.java.simpleName)
        et.onFocusChangeListener = View.OnFocusChangeListener { v, hasFocus -> tvFocus.text = "是否获得焦点:$hasFocus" }
        btnChangeDrawable.setOnClickListener {
            et.setClearTextDrawable(android.R.drawable.ic_menu_search)
        }
    }
}
